src/Hpp/CmdLine.hs

```diff
@@ -89,7 +89,7 @@ parseArgs cfg0 = go emptyEnv id cfg0 Nothing . concatMap breakEqs
           return . Left $ BadCommandLine "Multiple output files given"
 
 -- | Run Hpp with the given commandline arguments.
-runWithArgs :: [String] -> IO ()
+runWithArgs :: [String] -> IO (Maybe Error)
```

Comment:
We use this function directly, and being able to get the error was very useful. Probably a better way to do this is to add a new function with a new name.